### ISSUE TITLE
[Question] Does it support the combination of hubert-large + linear-projector + tinyllama

### ISSUE THREAD

**Issue by VictorChen2012** [ddlBoJack/SLAM-LLM#82]
### Motivation
Try to replicate your experiments in the paper. Due to resource constrain, I'd like to start from the combination `hubert-large + linear project + tiny-llama-1.1b-chat`.
 
### Error
I revised the examples/librispeech_asr/scripts/finetune_hubert_xtralarge_linear_vicuna_7b.sh,  specifying speech_encoder with hubert_large and llm with tiny-llama-chat respectively, and changed as well the encoder dim accordingly. 
Unfortunately, when training, an error arises as follows:

![企业微信截图_17162878152304](https://github.com/ddlBoJack/SLAM-LLM/assets/13539339/8aaf9889-9b31-46cf-bb70-aa496ccf3cf8)

In the slam_llm/models/slam_model.py, it seems that the length of modality_mask (i.e. audio_length) is shorter than the encoder outputs.
I only change the speech encoder here, so such an error shouldn't occur.  
From my understanding, the hubert-large and hubert-xtralarge share the same subsampling rate, so this change shouldn't affect the training.

### Question
Does the examples/librispeech_asr support the combination mentioned above? Thanks a lot!

**Comment by zszheng147** [comment]
Hi, thanks for pointing out this issue. Yes, we do support above combination. 
Please modify the `slam_model.py` or checkout our latest code of `slam_model.py`

`        if modality_mask is not None:
            modality_unmask_start_indices = (modality_mask == True).float().argmax(dim=1)
            modality_lengths = torch.clamp(modality_mask.sum(dim=1), max=encoder_outs.shape[1]).tolist()
            
            encoder_outs_pad = torch.zeros_like(inputs_embeds)
            for i in range(encoder_outs.shape[0]):
                encoder_outs_pad[
                    i, modality_unmask_start_indices[i] : modality_unmask_start_indices[i] + modality_lengths[i]
                ] = encoder_outs[i][:modality_lengths[i]]
            
            inputs_embeds = encoder_outs_pad + inputs_embeds * (~modality_mask[:, :, None])
`

**Comment by VictorChen2012** [comment]
> Hi, thanks for pointing out this issue. Yes, we do support above combination. Please modify the `slam_model.py` or checkout our latest code of `slam_model.py` (we will fix this in main branch soon).
> 
> ```
>         if modality_mask is not None:
>             modality_unmask_start_indices = (modality_mask == True).float().argmax(dim=1)
>             modality_lengths = torch.clamp(modality_mask.sum(dim=1), max=encoder_outs.shape[1]).tolist()
>             
>             encoder_outs_pad = torch.zeros_like(inputs_embeds)
>             for i in range(encoder_outs.shape[0]):
>                 encoder_outs_pad[
>                     i, modality_unmask_start_indices[i] : modality_unmask_start_indices[i] + modality_lengths[i]
>                 ] = encoder_outs[i][:modality_lengths[i]]
>             
>             inputs_embeds = encoder_outs_pad + inputs_embeds * (~modality_mask[:, :, None])
> ```

Problem solved, respect and thanks a lot!
